### DISCUSSION TITLE
[bug?] Rest params and dispatcher script

### DISCUSSION THREAD

**Discussion by ymeudal** [microcks/microcks#875] [Community]
Hello, 

I would like to report something strange that sounds as a bug. 
Let's say I have a rest operation, defined in postman by a GET {{baseurl}}/endpoint/p1/:param1

When I first upload my postman collection, I will have a default URI_PARTS dispatcher, with "param1" as the dispatching rule. 
All is good with this default setup, I am able to perform different GETs with my samples. 
Then, I would like to change dispatcher type to a SCRIPT, because I have to ehance some replies with some dynamic stuff, or get some extra query parameters. 
At this step, all is fine, script is doing what it is expected to do. The script is able to process even values of param1 not defined in samples.

I want to upload a second time my postman collection, for any reason. Script of the dispatcher remains, but it is now impossible to execute same requests: Error 404. 
All samples URLs shown in microcks under this operation are ending with /endpoint/p1/:param1 (instead of valued case like /endpoint/p1/chocolate ) - I suppose something is wrong with detecting URI pattern of the rest operation.

**Comment by lbroudoux** [comment]
Hey @ymeudal 
Thanks for this one.

I remember we had previous issue on same topic and it's possible some cases hadn't been fixed yet. Would you mind opening an issue in microcks repo so that we can track investigation and even fix on this one?

Thanks!

**Reply by ymeudal** [reply]
Hello @lbroudoux ,
You're right, this issue is nearly the same with this one: #742.
I created bug #877.

**Reply by lbroudoux** [reply]
I've now published a fix for this one. Let me know how things are going!

**Reply by ymeudal** [reply]
This bug has been fixed in 1.8.x branch, thank you.
